src/Provider/AbstractPasetoCookieSettingsProvider.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Helis\SettingsManagerBundle\Provider;
+
+abstract class AbstractPasetoCookieSettingsProvider extends AbstractCookieSettingsProvider
```

Comment:
In my opinion `AbstractCookieSettingsProvider` name no longer is accurate. Name suggests that it's for general cookie, however it's actualy used for cookies with PASETO encoding.
For this reason I've introduced `AbstractPasetoCookieSettingsProvider` and marked `AbstractCookieSettingsProvider` as deprecated. I couldn't think of better solution without introducing breaking changes.

Full cookie provider classes inheritance:

```
AbstractBaseCookieSettingsProvider
    AbstractPasetoCookieSettingsProvider (renamed deprecated AbstractCookieSettingsProvider)
        CookieSettingsProvider
        AsymmetricCookieSettingsProvider
    JwtCookieSettingsProvider
```

I even think it would make sense to do similar name changes to `CookieSettingsProvider -> PasetoCookieSettingsProvider` and `AsymmetricCookieSettingsProvider -> AsymmetricPasetoCookieSettingsProvider`.

Reply:
Ah yes, it has abstract methods with typehint objects from paseto. Name changes also make sense. Let's do those changes and deprecate old classes.

Reply:
Updated classes hierarchy:

```
AbstractBaseCookieSettingsProvider
    AbstractPasetoCookieSettingsProvider (renamed deprecated AbstractCookieSettingsProvider)
        PasetoCookieSettingsProvider (renamed deprecated CookieSettingsProvider)
        AsymmetricPasetoCookieSettingsProvider (renamed deprecated AsymmetricCookieSettingsProvider)
    JwtCookieSettingsProvider
```